src/TestProblem/Hydro/Riemann/Init_TestProb_Hydro_Riemann.cpp

```diff
@@ -425,7 +425,12 @@ void SetGridIC( real fluid[], const double x, const double y, const double z, co
                                     EoS_AuxArray_Flt, EoS_AuxArray_Int, h_EoS_Table );
 
 // do NOT include magnetic energy here
-   fluid[ENGY] = Hydro_ConEint2Etot( fluid[DENS], fluid[MOMX], fluid[MOMY], fluid[MOMZ], Eint, 0.0 );
+   fluid[ENGY   ] = Hydro_ConEint2Etot( fluid[DENS], fluid[MOMX], fluid[MOMY], fluid[MOMZ], Eint, 0.0 );
+#  ifdef TEMP_IG
+   fluid[TEMP_IG] = NULL_REAL;
```

Comment:
Remove this line.

Reply:
IIUC, @hfhsieh suggested me that we need to initialize `fluid[TEMP_IG]`  before invoking eos. 

https://github.com/hyschive/gamer-fork/pull/54#discussion_r993273477

Don't we need this?

Reply:
I see. Then keep it.

Reply:
My concern is that when the nuclear EoS fails, we will be able to trace it if `TEMP_IG` is a known value (`NULL_REAL`), rather than a random number.

Reply:
Good point. I forgot that `EoS_DensEint2Temp_CPUPtr()` also needs this temperature guess.